### ISSUE TITLE
GD2 smoke test failures

### ISSUE THREAD

**Issue by rishubhjain** [gluster/glusterd2#1428]
Reference https://build.gluster.org/job/gd2-smoke/3261/console

Failed to spawn glusterd2 in BrickMux e2e tests

Unable to reproduce this scenario locally.

**Comment by vpandey-RH** [comment]
@rishubhjain This happens sometimes. But, I have seen this error only during CI builds, Its very rare in my local as well. But retrigger the build and it will pass.

**Comment by rishubhjain** [comment]
@vpandey-RH Yes, but retriggering should not be a solution to the problem, trying to find out the real cause.

**Comment by aravindavk** [comment]
I suspect cleanup issue for connection refused errors. May be last tests didn't kill all running glusterd2 or something like that. Create a PR to Dump all glusterd2.log contents to stdout by adding logic in https://github.com/gluster/glusterd2/blob/master/extras/centos-ci.sh#L52  (Trigger retests to uncover the glusterd2 start issue)

**Comment by rishubhjain** [comment]
@aravindavk In brickmux e2e tests, the gd2 service is getting killed after setting up the cluster and then respawned, I will send the PR, it will give us a better insight.
